src/main/utils/mavenUtils.ts

```diff
@@ -0,0 +1,319 @@
+    for (let ProjectTree of prototypeTree) {
+        progress.report({ message: 'Analyzing pom.xml at ' + ProjectTree.pomPath });
+        // let projectDependenciesList: string[] = await getRawDependenciesList(ProjectTree.pomPath, treesManager);
+        // if (projectDependenciesList.length === 0) {
+        //     break;
+        // }
+        // FilterParentDependencies(projectDependenciesList);
+        let dependenciesTreeNode: MavenTreeNode = new MavenTreeNode(ProjectTree.pomPath, componentsToScan, treesManager, root);
+        dependenciesTreeNode.refreshDependencies(quickScan, ProjectTree, mavenTreeNodes);
+        mavenTreeNodes.push(dependenciesTreeNode);
+    }
+    return mavenTreeNodes;
+}
+
+/**
+ * @param pomArray list of all pom.xml uri inside root dir
+ * for each pom:
+ * 1. get the pomId(groupId,artifactId,version)
+ * 2. search pomId from step 1 in pomTree
+ *  2.1 if found remove from tree and otherwise create new node with pomId
+ * 3/ update the path/parent of node from step 3
+ * 4. try to add the node to its parent's children otherwise add it to the root of the tree.
+ */
+export function buildPrototypePomTree(pomArray: vscode.Uri[], treesManager: TreesManager): PomTree[] {
+    let result: PomTree[] = [];
+    pomArray.forEach(pom => {
+        try {
+            const [pomId, rawDependencies]: string[] = getPomDetails(pom.fsPath, treesManager);
+            if (!!pomId) {
+                let parentId: string = getParentInfo(pom);
+                let index: number = searchPomId(result, pomId);
+                let currNode: PomTree;
+                // If the node already exists get it and remove it from tree other wise create a new node
+                if (index > -1) {
+                    currNode = result[index];
+                    result.splice(index, 1);
+                } else {
+                    currNode = new PomTree(pomId);
+                }
+                currNode.pomPath = path.dirname(pom.fsPath);
+                currNode.rawDependencies = rawDependencies;
+                currNode.parentId = parentId;
+                addNode(result, currNode);
+            }
+        } catch (error) {
+            treesManager.logManager.logError(error, false);
+            treesManager.logManager.logMessage(
+                'Could not get dependencies tree from pom.xml (path: ' +
+                    pom +
+                    ').\n' +
+                    'Possible cause: The project needs to be installed by maven. Install it by running "mvn clean install" from ' +
+                    pom +
+                    '.',
+                'ERR'
+            );
+        }
+    });
+    return result;
+}
+// If the node have parent do:
+// 1. check if the parent already in the tree,
+//     1.1 if its add to its child
+//     1.2 else create the parent and add the parent to the array
+// 2.otherwise add the node to the array
+export function addNode(array: PomTree[], node: PomTree) {
```

Comment:
1. "array" is meaningless variable name.
2. Please put this functionality in PomTree class

Reply:
This function is not related to the PomTree class.
I've changed the name to `addPrototypeNode`